user/service.py

```diff
@@ -6,12 +6,17 @@
 import user.dao as user_dao
 import user_token.service as token_service
 from db_python_util.db_classes import User
+from db_python_util.db_exceptions import EntityNotFoundException
 
+import be_utilities.service_exceptions as service_exceptions
 def get(user_id: str = None) -> List[User] | Optional[User]:
     """Get a user by ID, or all users if no ID is given."""
 
     if user_id is not None:
-        return user_dao.get_by_id(user_id)
+        try:
+            return user_dao.get_by_id(user_id)
+        except EntityNotFoundException as e:
+            raise service_exceptions.MissingException(f"The given user ID '{user_id}' does not correspond to an existing user.")
```

Comment:
for the record e should have a message property with the same sort of message that you can pass along if you like